openexp/_canvas/xpyriment.py

```diff
@@ -107,14 +107,8 @@ def show(self):
 		return self.experiment.time()
 
 	@configurable
-	def clear(self, color=None):
-
-		if color is not None:
-			if u'color' in cfg:
-				warnings.warn(u'color is a deprecated style argument for '
-					'canvas.clear(). Use background_color instead.',
-					DeprecationWarning)
-			self.background_color = color
+	def clear(self):
```

Comment:
There's a bug here. It's actually not your fault (it was already there), but this may nevertheless be a good moment to fix it (and a good opportunity to practice). Take this script:

~~~ .python
k = keyboard()
# Create an show a red canvas
c = canvas(background_color='red')
c.show()
# Wait
k.get_key()
# Clear to blue. Doesn't work!
c.clear(background_color='blue')
c.show()
# Wait
k.get_key()
~~~

This should show a red screen, followed by a blue screen. But with the *xpyriment* backend this doesn't happen: The blue background isn't applied. The reason is that the canvas is already prepared, and therefore the new background color doesn't have any effect. If you look at the `xpyriment.add_stim()` function you see the correct logic, which you can basically copy (with the exception of the `prepare` keyword, which `clear()` doesn't have):

- Make the changes, in this case emptying the `stim_list`
- Set prepared to False
- If auto_prepare is True, then call prepare()

Do you think you can fix that? :-)